cylc/flow/config_schema.py

```diff
@@ -17,284 +17,451 @@
-            'inactivity handler': [VDR.V_STRING_LIST, None],
-            'shutdown handler': [VDR.V_STRING_LIST, None],
-            'aborted handler': [VDR.V_STRING_LIST, None],
-            'stalled handler': [VDR.V_STRING_LIST, None],
-            'timeout': [VDR.V_INTERVAL],
-            'inactivity': [VDR.V_INTERVAL],
-            'abort if startup handler fails': [VDR.V_BOOLEAN],
-            'abort if shutdown handler fails': [VDR.V_BOOLEAN],
-            'abort if timeout handler fails': [VDR.V_BOOLEAN],
-            'abort if inactivity handler fails': [VDR.V_BOOLEAN],
-            'abort if stalled handler fails': [VDR.V_BOOLEAN],
-            'abort if any task fails': [VDR.V_BOOLEAN],
-            'abort on stalled': [VDR.V_BOOLEAN],
-            'abort on timeout': [VDR.V_BOOLEAN],
-            'abort on inactivity': [VDR.V_BOOLEAN],
-            'mail events': [VDR.V_STRING_LIST, None],
-            'mail from': [VDR.V_STRING],
-            'mail smtp': [VDR.V_STRING],
-            'mail to': [VDR.V_STRING],
-            'mail footer': [VDR.V_STRING],
-        },
-        'reference test': {
-            'expected task failures': [VDR.V_STRING_LIST],
-        },
-        'authentication': {
-            # Allow owners to grant public shutdown rights at the most, not
-            # full control.
-            'public': (
-                [VDR.V_STRING, '']
-                + [level.name.lower().replace('_', '-') for level in [
-                    Priv.IDENTITY, Priv.DESCRIPTION, Priv.STATE_TOTALS,
-                    Priv.READ, Priv.SHUTDOWN]])
-        },
-    },
-    'scheduling': {
-        'initial cycle point': [VDR.V_CYCLE_POINT],
-        'final cycle point': [VDR.V_STRING],
-        'initial cycle point constraints': [VDR.V_STRING_LIST],
-        'final cycle point constraints': [VDR.V_STRING_LIST],
-        'hold after point': [VDR.V_CYCLE_POINT],
-        'cycling mode': (
-            [VDR.V_STRING, Calendar.MODE_GREGORIAN] +
-            list(Calendar.MODES) + ["integer"]),
-        'runahead limit': [VDR.V_STRING],
-        'max active cycle points': [VDR.V_INTEGER, 3],
-        'spawn to max active cycle points': [VDR.V_BOOLEAN],
-        'queues': {
-            'default': {
-                'limit': [VDR.V_INTEGER, 0],
-                'members': [VDR.V_STRING_LIST],
-            },
-            '__MANY__': {
-                'limit': [VDR.V_INTEGER, 0],
-                'members': [VDR.V_STRING_LIST],
-            },
-        },
-        'special tasks': {
-            'clock-trigger': [VDR.V_STRING_LIST],
-            'external-trigger': [VDR.V_STRING_LIST],
-            'clock-expire': [VDR.V_STRING_LIST],
-            'sequential': [VDR.V_STRING_LIST],
-            'exclude at start-up': [VDR.V_STRING_LIST],
-            'include at start-up': [VDR.V_STRING_LIST],
-        },
-        'xtriggers': {
-            '__MANY__': [VDR.V_XTRIGGER],
+
+
+def upg(cfg, descr):
+    SPEC = {
```

Comment:
Why has the `SPEC` been moved inside the `upg` function?

Reply:
This is a mistake.